tests/components/tedee/test_sensor.py

```diff
@@ -0,0 +1,56 @@
+"""Tests for the Tedee Sensors."""
+
+
+from unittest.mock import MagicMock
+
+import pytest
+from syrupy import SnapshotAssertion
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+pytestmark = pytest.mark.usefixtures("init_integration")
+
+
+async def test_battery(
+    hass: HomeAssistant,
+    mock_tedee: MagicMock,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test tedee battery sensor."""
+    state = hass.states.get("sensor.lock_1a2b_battery")
+    assert state
+    assert state == snapshot
+
+    entry = entity_registry.async_get(state.entity_id)
+    assert entry
+    assert entry.device_id
+    assert entry == snapshot
```

Comment:
You could also iterate over them, checkout the withings sensor tests

Reply:
Or at least, you dont have to have a separate test for every entity, you could also merge these two and use a for loop (since the withings way might be too overkill for 2 entities)